src/commonMain/kotlin/data/buffs/BadgeOfTheSwarmguard.kt

```diff
@@ -0,0 +1,81 @@
+package data.buffs
+
+import character.Ability
+import character.Buff
+import character.Proc
+import character.Stats
+import data.model.Item
+import sim.Event
+import sim.SimParticipant
+
+class BadgeOfTheSwarmguard : Buff() {
+    companion object {
+        const val name = "Badge of the Swarmguard (static)"
+    }
+    override val id: Int = 26480
+    override val name: String = Companion.name
+    override val durationMs: Int = -1
+    override val hidden: Boolean = true
+
+    val buffDurationMs = 30000
+    val buff = object : Buff() {
+        override val name: String  = "Badge of the Swarmguard"
+        override val durationMs: Int = buffDurationMs
+
+        val stackBuff = object : Buff() {
+            override val name: String = "Insight of the Qiraji"
+            override val durationMs: Int = -1
```

Comment:
I'm not sure this would ever be removed, so stacks on repeated Badge activations might remember the previous stack state (and be fully-stacked instantly).  With the suggested refactor above, we shouldn't have to worry about that since the single Buff will have a duration, and be auto-removed on expiration.